whereby/sdk
# Add lock/unlock room functionality

### Description

Add `roomKey`-based authentication controls and allow participants joining with the `host` role to lock/unlock rooms.

**Summary:**



**Related Issue:**



### Testing

1. Check out this branch
2. In the `/.env` file add a valid _host_ `roomKey` for the configured `STORYBOOK_ROOM` in following line:

```
STORYBOOK_ROOM_HOST_ROOMKEY=[host_roomKey_here]
```

3. Run `yarn build && yarn dev`
4. In the Storybook UI (@ http://localhost:6006/) open the story: "Custom UI" > "Room Connection With Host Controls"
5. Check that the `roomKey` is filled with the host room key added in step 3.
6. Clicking on the lock/unlock buttons should now perform the action in the connected room [Screenshot 1].
7. In the Story controls panel, remove the roomKey and reload the page (Cmd + R).
8. The lock/unlock controls shown in the UI should now be in red [Screenshot 2]
9. Clicking on the lock/unlock buttons should not perform the action in the connected room (not connected as a "host" now).
10. Additionally, a warning should be displayed in the console whenever these actions are invoked by a non-"host" user.

### Screenshots/GIFs (if applicable)

#### Screenshot 1: "host" `roomKey` provided

<img width="751" alt="Screenshot 2024-04-05 at 14 37 10" src="https://github.com/whereby/sdk/assets/119658286/a0313030-e63b-4040-8467-bc5f672a6d43">

#### Screenshot 2: "host" `roomKey` not provided

<img width="756" alt="Screenshot 2024-04-05 at 14 38 52" src="https://github.com/whereby/sdk/assets/119658286/3ed3fb20-74b6-4c32-b5c9-18d12128ce1d">

### Checklist

        under the terms of the MIT license.
